### PR TITLE
Consolidates CI

### PR DESCRIPTION
**Pull request by pedrocamargo**
Consolidates CI workflows using appropriate workflow context. Resulting changes:
- Reduces workflows from 14 to 5
- Windows testing expanded to 32 bits
- Windows testing only performed after linting succeeds

### CODE REVIEW

**Comment by pedrocamargo** [comment]
Now I think this is as clean as it should be, @jamiecook

File: .github/workflows/build_mac.yml [hunk 1]
```diff
@@ -22,15 +19,23 @@ jobs:
...
         pip install setuptools wheel twine
-        pip install numpy cython pyaml pyqt5 pyshp scipy
+        pip install -r requirements.txt
 
     - name: Build
       run: |
         python setup.py sdist bdist_wheel
 
     - name: Publish
+      if: ${{ github.event_name == 'release'}}
```
Comment (jamiecook) [inline .github/workflows/build_mac.yml:29]: Thats a nice change :)

File: .github/workflows/documentation.yml [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-  release:
-    types: [prereleased, published]
+on:  [pull_request, release]
 
 jobs:
   build:
     runs-on: ubuntu-latest
     strategy:
-      max-parallel: 4
+      max-parallel: 1
```
Comment (jamiecook) [inline .github/workflows/documentation.yml:9]: no parallels?

**Review by jamiecook — APPROVED** [review]

File: .github/workflows/documentation.yml (hunk 2 above)
Comment (pedrocamargo) [reply .github/workflows/documentation.yml:9]: No need

**Pull request merged**